layers/meta-resin-edison/recipes-bsp/u-boot/files/0001-u-boot-env-Add-the-ability-to-merge-the-saved-env-with-the-default.patch

```diff
@@ -0,0 +1,34 @@
+From: Theodor Gherzan <theodor@resin.io>
+Date: Mon, 14 Aug 2017 15:51:09 +0200
+Subject: [PATCH] Merge default environment with the saved one
+
+When u-boot imports an environment it overwrites the default one compiled inside
+the u-boot binary. We need these two environments to be merged, so we can expose
+the resin specific definition from resin-uboot to the environment.
+
+Signed-off-by: Theodor Gherzan <theodor@resin.io>
+
+Upstream-Status: Innapropriate
+---
+ common/env_common.c | 18 ++++++++++++++++++
+ 1 file changed, 18 insertions(+)
+
+diff --git a/common/env_common.c b/common/env_common.c
+index af59c72..c8b8cfc 100644
+--- a/common/env_common.c
++++ b/common/env_common.c
+@@ -215,6 +228,11 @@ int env_import(const char *buf, int check)
+ 	if (himport_r(&env_htab, (char *)ep->data, ENV_SIZE, '\0', 0, 0,
```

Comment:
Redo this patch with the following changes please:

-       if (himport_r(&env_htab, (char *)ep->data, ENV_SIZE, '\0', 0, 0,
+       set_default_env(NULL);
+       if (himport_r(&env_htab, (char *)ep->data, ENV_SIZE, '\0', H_NOCLEAR, 0,